packages/image_picker/ios/Classes/ImagePickerPlugin.m

```diff
@@ -69,6 +73,33 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
                                    details:nil]);
         break;
     }
+  } else if ([@"pickVideo" isEqualToString:call.method]) {
+    _imagePickerController.modalPresentationStyle = UIModalPresentationCurrentContext;
+    _imagePickerController.delegate = self;
+    _imagePickerController.mediaTypes = @[
+      (NSString *)kUTTypeMovie, (NSString *)kUTTypeAVIMovie, (NSString *)kUTTypeVideo,
+      (NSString *)kUTTypeMPEG4
+    ];
+    _imagePickerController.videoQuality = UIImagePickerControllerQualityTypeHigh;
+
+    _result = result;
```

Comment:
The code from here and until line 102 appears to be a copy of the image case. Could we refactor to avoid the duplication?

Reply:
Would a shared function containing the shared code be better for this? I duplicated it so I could provide different errors in the result.

Reply:
I'll leave it up to you.